generators/service-watson-discovery/templates/python/instrumentation.py

```diff
@@ -1,10 +1,28 @@
 from ibmcloudenv import IBMCloudEnv
 from watson_developer_cloud import DiscoveryV1
 
-discovery = DiscoveryV1(
-    username=IBMCloudEnv.getString('watson_discovery_username'),
-    password=IBMCloudEnv.getString('watson_discovery_password'),
-    version='2016-12-01')
+params = {
+    version: '2018-03-05',
+    url: IBMCloudEnv.getString('watson_discovery_url')
+}
+if IBMCloudEnv.getString('watson_discovery_apikey'):
+    iam_url = 'https://iam.stage1.bluemix.net/identity/token' if 'gateway-s.' in params.url else 'https://iam.bluemix.net/identity/token'
+    iam_apikey = api_key=IBMCloudEnv.getString('watson_discovery_apikey')
 
+    discovery = DiscoveryV1(
+        url=IBMCloudEnv.getString('watson_discovery_url'),
+        iam_api_key=iam_apikey,
```

Comment:
provide `iam_url` to make sure it works in staging:

```
DiscoveryV1(
       url=IBMCloudEnv.getString('watson_discovery_url'),
       iam_api_key=iam_apikey,
       iam_url=iam_url,
       version='2018-03-05')
```

`iam_url` is optional and SDK defaults to use the production one, but for staging we need to provide one. See https://github.com/watson-developer-cloud/python-sdk/#supplying-the-iam-api-key